apps/doji-backend/src/session/session.service.ts

```diff
@@ -81,8 +85,39 @@ export class SessionService {
       throw new NotFoundException('Session not found or you are not in the shcedule')
     }
     user.sessions.remove(session)
-    session.participants.remove(user)
+    session.participants.remove(user as User)
     this.sessionRepo.flush()
     return
   }
+
+  async getSessionInfo(id: number): Promise<Session | null> {
+    return this.sessionRepo.findOne({ id: id }, ['reviews', 'reviews.user'])
+  }
+
+  async calculateReviewStatForSession(session: Session): Promise<IReviewStat> {
+    const counts: { rating: number; count: string }[] = await this.em
+      .createQueryBuilder(Session, 's')
+      .select(['r.rating', 'count(*)'])
```

Comment:
I wonder why you don't calculate avg using `AVG(*)` here?

Reply:
Why should that be done?

Reply:
nvm. I'm just curious

Reply:
I do this because it simplify unmarshalling code.